gremlin_client.go

```diff
@@ -120,14 +120,10 @@ func (c *GremlinClient) execWithRetry(ctx context.Context, query string, queryId
 			if err != nil {
 				return nil, err
 			}
+			defer lock.Unlock()
```

Comment:
you don't wanna defer in a potential loop, and i think it was fine as it was tbh